### PR TITLE
Adding a test to make sure Puts with timestamps are retryable.

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java [hunk 1]
```diff
@@ -122,13 +121,18 @@
...
-    SslContextBuilder sslBuilder = GrpcSslContexts.forClient();
-    sslBuilder.ciphers(null);
-    return sslBuilder;
+    try {
+      SslContextBuilder sslBuilder = GrpcSslContexts.forClient();
+      sslBuilder.ciphers(null);
+      return sslBuilder;
+    } catch (IllegalArgumentException e) {
+      LOG.error("Could not initialize GrpcSslContexts", e);
+      return null;
```
Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: How is this null value handled when it is returned from here?

**Comment by sduskis** [comment]
I updated the BigtableSession sslBuilder handling to be a bit more in line with what we actually need.  We needed to bypass the sslBuilder creation until we needed it to create a BigtableSession rather than at class loading time.  There were failing unit tests that were testing a different aspect of Bigtable Session, but were failing on the creation of the sslBuilder, since ALPN was not present.  The new getSslBuilder() method defers the construction of the sslBuilder for that purpose.

PTAL

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java [hunk 2]
```diff
@@ -127,8 +126,16 @@ private static SslContextBuilder createGrpcSslBuilder() {
...
+    if (sslBuilder == null) {
+      sslBuilder = createGrpcSslBuilder();
+    }
+    return sslBuilder;
+  }
+
   private static SslContext createSslContext() throws SSLException {
-     return sslBuilder.build();
+    SslContextBuilder builder = getSslBuilder();
+    return builder == null ? null : builder.build();
```
Comment (kevinsi4508) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: can builder even be null?

Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java]: You're right.  I'll inline it.

**Comment by sduskis** [comment]
I inlined getSslBuilder().

File: bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/TestPutAdapter.java [hunk 3]
```diff
@@ -175,4 +178,23 @@ public void testEmptyPut() {
...
+    byte[] family1 = dataHelper.randomData("f1");
+    byte[] qualifier1 = dataHelper.randomData("qual1");
+    byte[] value1 = dataHelper.randomData("v1");
+
+    Put hbasePut = new Put(row, System.currentTimeMillis());
+    hbasePut.addColumn(family1, qualifier1, value1);
+    MutateRowRequest.Builder rowMutationBuilder = adapter.adapt(hbasePut);
+    MutateRowRequest request = rowMutationBuilder.build();
+    Predicate predicate =
+        BigtableSession.METHODS_TO_RETRY_MAP.get(BigtableServiceGrpc.METHOD_MUTATE_ROW);
```
Comment (kevinsi4508) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/TestPutAdapter.java]: can you verify all predicates in the map and with different inputs?

Comment (sduskis) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/TestPutAdapter.java]: Not in TestPutAdapter.  Other methods take different input.

**Comment by kevinsi4508** [comment]
LGTM

**Comment by derjust** [comment]
Highly promoting here to add a large notice in the README.md/java docs that it is (nearly) always preferably to use `new Put(..., timestamp)` over `new Put(...)` - From the (HBase?) API docs it just reads that overriding the timestamp has only an impact on the versioning and not around retry/reliability.

**Pull request merged**